### PR TITLE
remove ability to propagate baggage on its own to avoid app crashes

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
Fixes this [issue](https://github.com/DataDog/dd-trace-js/issues/5111) at the expense of not allowing customers to propagate baggage on its own.
This is a quick fix for now to avoid crashing customer applications, we will add a more involved solution to provide customers with the full functionality of baggage in a later PR.

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/opentracing/propagation/text_map.js [hunk 1]
```diff
@@ -325,6 +325,9 @@ class TextMapPropagator {
       if (context === null) {
         context = extractedContext
         if (this._config.tracePropagationExtractFirst) {
+          if (this._hasPropagationStyle('extract', 'baggage') && carrier.baggage) {
+            if (context) this._extractBaggageItems(carrier, context)
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/propagation/text_map.js:329]: Since the check for `context` is now always needed, let's move it to `_extractBaggageItems` instead of nesting conditions.

File: packages/dd-trace/test/opentracing/propagation/text_map.spec.js [hunk 2]
```diff
@@ -451,7 +451,7 @@ describe('TextMapPropagator', () => {
       expect(spanContextD._baggageItems).to.deep.equal({})
     })
 
-    it('should extract baggage when it is the only propagation style', () => {
+    it('should not extract baggage when it is the only propagation style', () => {
```
Comment (rochdev) [inline packages/dd-trace/test/opentracing/propagation/text_map.spec.js:454]: Let's skip the test instead since the expected behaviour is not to skip extraction, it's just a temporary limitation.

Comment (ida613) [reply packages/dd-trace/test/opentracing/propagation/text_map.spec.js:454]: @bm1549 what do you think?

Comment (bm1549) [reply packages/dd-trace/test/opentracing/propagation/text_map.spec.js:454]: More test coverage is nearly always better imo

What about if we just add a comment on top that describes that this is because of a temporary limitation? Alternatively, we can comment the test if we really don't want to run it, but that's my least preferred option

@rochdev - if none of that is acceptable we can delete the test

Comment (rochdev) [reply packages/dd-trace/test/opentracing/propagation/text_map.spec.js:454]: > What about if we just add a comment on top that describes that this is because of a temporary limitation?

Good enough, I just want to avoid a situation where adding the feature would fail a test, and then a reviewer (or other implementor) is not sure if they're breaking something that was supposed to work differently (because of a test testing a bug)

Comment (bm1549) [reply packages/dd-trace/test/opentracing/propagation/text_map.spec.js:454]: On a similar note, I wanted to avoid the situation where we re-introduce the bug without fixing the edge case and not see this test failing to stop us

File: packages/dd-trace/src/opentracing/propagation/text_map.js (hunk 1 above)
Comment (ida613) [reply packages/dd-trace/src/opentracing/propagation/text_map.js]: updated

File: packages/dd-trace/test/opentracing/propagation/text_map.spec.js (hunk 2 above)
Comment (ida613) [reply packages/dd-trace/test/opentracing/propagation/text_map.spec.js:455]: added a comment

**Review by bm1549 — DISMISSED** [review]

File: packages/dd-trace/src/opentracing/propagation/text_map.js [hunk 3]
```diff
@@ -596,6 +598,7 @@ class TextMapPropagator {
   }
 
   _extractBaggageItems (carrier, spanContext) {
+    if (!carrier) return
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/propagation/text_map.js:601]: Shouldn't this be `!spanContext`?

File: packages/dd-trace/src/opentracing/propagation/text_map.js [hunk 4]
```diff
@@ -325,6 +325,9 @@ class TextMapPropagator {
       if (context === null) {
         context = extractedContext
         if (this._config.tracePropagationExtractFirst) {
+          if (this._hasPropagationStyle('extract', 'baggage') && carrier.baggage) {
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/propagation/text_map.js:328]: I just noticed that this condition could also be moved to `_extractBaggageItems`.

**Review by rochdev — APPROVED** [review]
